src/tree.rs

```diff
@@ -732,3 +842,65 @@ const _: () = {
         }
     }
 };
+
+#[cfg(test)]
+mod test {
+    use super::{normalize_params, Node};
+
+    #[test]
+    fn test_normalization() {
+        let path = "/users/:azda/test/:test";
+        let (a, b) = normalize_params(path.to_string().into_bytes()).unwrap();
+        let a1 = std::str::from_utf8(&a).unwrap();
+        let b1 = std::str::from_utf8(&b[0]).unwrap();
+        let b2 = std::str::from_utf8(&b[1]).unwrap();
+        dbg!(a1, b1, b2);
+    }
+
+    #[test]
+    fn test_remove() {
+        const ROUTES: &[&str] = &[
+            "/hi",
+            "/contact",
+            "/co",
+            "/c",
+            "/a",
+            "/ab",
+            "/doc/",
+            "/doc/rust_faq.html",
+            "/doc/rust1.26.html",
+            "/ʯ",
+            "/β",
+            "/sd!here",
+            "/sd$here",
+            "/sd&here",
+            "/sd'here",
+            "/sd(here",
+            "/sd)here",
+            "/sd+here",
+            "/sd,here",
+            "/sd;here",
+            "/sd=here",
+            "/get/test/abc",
+            "/get/:param/abc",
+            "/something/:pzd/thirdthing",
+            "/get/abc",
+            "/get/:param",
+            "/foo/*catchall",
+            "/bar",
+            "/bar/",
+            "/bar/*catchall",
+        ];
+        let mut root = Node::default();
+
+        for route in ROUTES {
+            root.insert(*route, ()).unwrap();
+        }
+
+        println!("{root}");
```

Comment:
Can you add tests that make sure the tree stays consistent after removal? i.e try matching against related routes after a removal.

(Also avoid the debug/print statements in the tests).